web/src/component/access/AssetTree.js

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 The casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, {useEffect, useMemo, useState} from "react";
+import {Tree} from "antd";
+import * as AssetBackend from "../../backend/AssetBackend";
+import * as Setting from "../../Setting";
+import Search from "antd/es/input/Search";
+
+const AssetTree = ({onSelect, account}) => {
+  const [expandedKeys, setExpandedKeys] = useState(["windows", "linux"]);
+  const [treeData, setTreeData] = useState([]);
+  const [searchValue, setSearchValue] = useState("");
+  const [autoExpandParent, setAutoExpandParent] = useState(true);
+
+  useEffect(() => {
+    AssetBackend.getAssets(account.owner, "", "", "name", account.name)
+      .then((res) => {
+        if (res.status === "ok") {
+          setTreeData(transformData(res.data));
+        } else {
+          Setting.showMessage("error", `Failed to get assets: ${res.msg}`);
+        }
+      })
+      .catch((error) => {
+        Setting.showMessage("error", `Failed to get assets: ${error}`);
+      });
+  }, []);
+
+  const onExpand = (newExpandedKeys) => {
+    setExpandedKeys(newExpandedKeys);
+    setAutoExpandParent(false);
+  };
+
+  const onChange = (e) => {
+    const {value} = e.target;
+    setSearchValue(value);
+    setAutoExpandParent(true);
+  };
+
+  const handleNodeSelect = (selectedKeys) => {
+    if (selectedKeys.length > 0) {
+      onSelect(selectedKeys[0]);
+    }
+  };
+
+  const transformData = (data) => {
+    const linuxAssets = [];
+    const windowsAssets = [];
+
+    data.forEach((asset) => {
+      const {protocol, port} = asset;
+
+      if (protocol === "ssh" && port === 22) {
+        linuxAssets.push({
```

Comment:
Auto-grouping by Windows and Linux is not very valuable. Using icon to differentiate OS is better. Grouping and orders should be customized by user